src/App.js

```diff
@@ -1,10 +1,145 @@
+    const sumLessThanMax = allAttrs.reduce((acc, curr) => acc + curr) <= maxAttrSum;
+
+    return sumLessThanMax && !allAttrs.find((attr) => attr < 0 || attr > maxAttr);
+  }
+
+  validateForm = () => {
+    const {
+      formData: {
+        cardName,
+        cardDescription,
+        cardImage,
+        cardRare,
+      },
+    } = this.state;
+    const isSaveButtonDisabled = !(
+      cardName && cardDescription && cardImage && cardRare && this.validateAttr()
+    );
+    this.setState({
+      isSaveButtonDisabled,
+    });
+  }
+
+  onInputChange = ({ target: { value, id, type, checked } }) => {
+    this.setState(({ formData }) => ({
+      formData: { ...formData, [id]: type === 'checkbox' ? checked : value },
+    }), this.validateForm);
+  }
+
+  onSaveButtonClick = () => {
+    this.setState(({ cards, formData, hasTrunfo }) => ({
+      cards: [...cards, formData],
+      hasTrunfo: hasTrunfo || formData.cardTrunfo,
+      formData: INITIAL_FORM_STATE,
+    }));
+  }
+
+  deleteCard = (index) => {
+    const { cards, hasTrunfo } = this.state;
+    const removedCard = cards.splice(index, 1);
+    this.setState({
+      cards,
+      hasTrunfo: !(removedCard[0].cardTrunfo && hasTrunfo),
+    });
+  }
+
   render() {
+    const {
+      formData,
+      isSaveButtonDisabled,
+      hasTrunfo,
+      cards,
+      filter,
+      rareFilter,
+      trunfoFilter,
+    } = this.state;
     return (
       <div>
         <h1>TrybeTrunfo</h1>
+        <Form
+          onInputChange={ this.onInputChange }
+          onSaveButtonClick={ this.onSaveButtonClick }
+          isSaveButtonDisabled={ isSaveButtonDisabled }
+          hasTrunfo={ hasTrunfo }
+          { ...formData }
+        />
+        <Card { ...formData } />
+        <input
+          type="text"
+          data-testid="name-filter"
+          onChange={ ({ target }) => this.setState({ filter: target.value }) }
+        />
+        <select
+          data-testid="rare-filter"
+          onChange={ ({ target }) => this.setState({ rareFilter: target.value }) }
+        >
+          <option value="todas">Todas</option>
+          <option value="normal">Normal</option>
+          <option value="raro">Raro</option>
+          <option value="muito raro">Muito Raro</option>
+        </select>
+        <input
+          type="checkbox"
+          data-testid="trunfo-filter"
+          onChange={ ({ target }) => this.setState({ trunfoFilter: target.checked }) }
+        />
+        {cards.filter((card) => {
+          const { cardName, cardRare, cardTrunfo } = card;
+          const nameFiltered = cardName.toLowerCase().includes(filter.toLowerCase());
+          const rareFiltered = cardRare === rareFilter || rareFilter === 'todas';
+          const trunfoFiltered = (trunfoFilter && cardTrunfo) || !trunfoFilter;
+          return nameFiltered && rareFiltered && trunfoFiltered;
+        }).map((card, index) => (
+          <>
```

Comment:
@EduardoSantosF como levantei na thread do Slack, uma das possíveis razões de o teste 12 estar quebrando é porque nesse loop não temos um elemento com uma `key` que englobe todos os elementos, aí a RTL acaba se perdendo na referência dos objetos. Alterando esse ponto deve resolver o problema.
```suggestion
          <div key={card.cardName}>
```

Reply:
Fiz essa modificação, isso realmente poderia dar em algum erro em outro momento, mas esse não foi o caso, o requisito 12 continua não passando. @renatofreire